checkout/access-checkout-android/src/main/java/com/worldpay/access/checkout/CardListener.kt

```diff
@@ -4,16 +4,81 @@ import android.text.InputFilter
 import com.worldpay.access.checkout.model.CardBrand
 import com.worldpay.access.checkout.validation.ValidationResult
 
+/**
+ * The purpose of this interface is that it will receive updates of [ValidationResult]'s and [CardBrand]'s from the [Card]
+ * when the fields are being edited or have completed editing. Implementations of this interface can decide how to apply
+ * those validation results to the card fields or any other appropriate UI elements in the application
+ *
+ * A [ValidationResult] can either be partially and/or completely valid and/or invalid. Implementers can choose how to style
+ * their payment forms based on the validation state of the fields at the moment the user is filling out their card data
+ *
+ * A [CardBrand] is an object that represents the card scheme that is linked to the pan that is being entered by the user.
+ * Implementers can choose how to style their pan field based on this information, such as showing a small icon for that
+ * card brand.
+ *
+ * A [InputFilter] can be attached to a field to restrict the length of that field based on the current input.
+ * For example, a card brand may have a particular limit on the number of characters it will accept and therefore applying the length
+ * filter will help reduce human user error on that field by not allowing them to type more characters than the maximum
+ * length for that card brand
+ */
 interface CardListener {
-    fun onPANUpdateValidationResult(validationResult: ValidationResult, cardBrand: CardBrand?, lengthFilter: InputFilter?)
-    fun onPANEndUpdateValidationResult(validationResult: ValidationResult, cardBrand: CardBrand?)
 
-    fun onCVVUpdateValidationResult(validationResult: ValidationResult, lengthFilter: InputFilter?)
-    fun onCVVEndUpdateValidationResult(validationResult: ValidationResult)
+  /**
+   * Receives an update once any validation has finished after an update to the pan field
+   *
+   * @param validationResult The validation result of this pan field update
+   * @param cardBrand (Optional) The card brand that was associated with this pan field update
```

Comment:
* @param cardBrand (Optional) The card brand, if identified, that is associated with this pan field update